### ISSUE TITLE
OpenAPI Spec - define an error structure

### ISSUE THREAD

**Issue by sebastian-bury** [Unleash/unleash#2974]
### Describe the feature request

Currently the specification does not provide an error structure. An error structure would help be able to gracefully and better handle errors if they are seen.

**Comment by sebastian-bury** [comment]
Additional notes: 

Should have at least one error structure at paths:
                        - paths./api/client/features.get
                        - paths./api/client/features/{featureName}.get
                        - paths./api/client/[metrics.post](http://metrics.post/)
                        - paths./api/client/[register.post](http://register.post/)
                        - paths./health.get

Missing error schema at path:
                        - paths./api/client/metrics.post.responses.400

Should have required property 'errors' at path:
                        - paths./health.get.responses.500.schema.properties

**Comment by thomasheartman** [comment]
@sebastian-bury Just out of curiosity: where did you get the data for the additional notes? Why should all those endpoints have at least one error structure etc? Not saying it's wrong at all, just curious where that came from ☺️

**Comment by gastonfournier** [comment]
Just double checked and these are still not having errors:
- https://docs.getunleash.io/reference/api/unleash/get-client-feature 
- https://docs.getunleash.io/reference/api/unleash/get-all-client-features
- https://docs.getunleash.io/reference/api/unleash/register-client-application
- https://docs.getunleash.io/reference/api/unleash/register-client-metrics (this one declares a 400 error, but maybe there are other errors as well)
- https://docs.getunleash.io/reference/api/unleash/get-health (this one declares a 500, but maybe there are other errors as well)

**Comment by sebastian-bury** [comment]
@gastonfournier this came from an enterprise customer, DM me for the name, I would assume all endpoints could have errors such as unauthorized?

**Comment by gastonfournier** [comment]
> @gastonfournier this came from an enterprise customer, DM me for the name, I would assume all endpoints could have errors such as unauthorized?

Yes, that's why I kept this open with the list of endpoints. We need to prioritize this if this is important

**Comment by sebastian-bury** [comment]
> Yes, that's why I kept this open with the list of endpoints. We need to prioritize this if this is important

not important enough to prioritize currently, they haven't raised it recently
